_inc/lib/components.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Components Library
+ *
+ * Load and display a pre-rendered component
+ */
+class Jetpack_Components {
+	/**
+	 * Load and display a pre-rendered component
+	 *
+	 * @since 7.6.0
+	 *
+	 * @return string The component markup
+	 */
+	public static function render_component( $name, $props ) {
+		wp_enqueue_style( 'wp-components' );
+
+		$rtl = is_rtl() ? '.rtl' : '';
+		wp_enqueue_style( 'jetpack-components', plugins_url( "_inc/blocks/components{$rtl}.css", JETPACK__PLUGIN_FILE ), array(), JETPACK__VERSION );
+
+		ob_start();
+		// `include` fails gracefully and throws a warning, but doesn't halt execution.
+		include JETPACK__PLUGIN_DIR . "_inc/blocks/$name.html";
+		$markup = ob_get_clean();
```

Comment:
BTW just curious, why not [`readfile()`](https://www.php.net/manual/en/function.readfile.php)?

Reply:
'cause we're actually executing it -- remember that it contains `<?php __( 'Some translated text', 'jetpack' ) ?>` expressions :sunglasses: